test/extended/router/h2spec.go

```diff
@@ -0,0 +1,289 @@
+						"http2/6.9.1.3":   true,
+						"http2/8.1.2.1.1": true,
+						"http2/8.1.2.1.2": true,
+						"http2/8.1.2.1.4": true,
+						"http2/8.1.2.2.1": true,
+						"http2/8.1.2.2.2": true,
+						"http2/8.1.2.3.1": true,
+						"http2/8.1.2.3.2": true,
+						"http2/8.1.2.3.3": true,
+						"http2/8.1.2.3.4": true,
+						"http2/8.1.2.3.5": true,
+						"http2/8.1.2.3.6": true,
+						"http2/8.1.2.3.7": true,
+						"http2/8.1.2.6.1": true,
+						"http2/8.1.2.6.2": true,
+					},
+				},
+				{
+					routeType: "reencrypt",
+					knownFailures: map[string]bool{
+						"http2/4.2.3": true,
+					},
+				},
+			}
+
+			for _, t := range routeTypeTests {
+				if t.routeType == "edge" {
+					e2e.Logf("Skipping %q tests; waiting for: https://github.com/openshift/router/pull/104", t.routeType)
+					continue
+				}
+				if t.routeType == "passthrough" {
+					// TODO(frobware): work out why the underlying nginx config errors so much
+					e2e.Logf("Skipping %q tests", t.routeType)
+					continue
+				}
+
+				t.hostname = getHostnameForRoute(oc, fmt.Sprintf("h2spec-nginx-%s", t.routeType))
+				g.By(fmt.Sprintf("[%s] Running conformance tests against host %q", t.routeType, t.hostname))
+
+				// sampleForFailures when true runs
+				// the conformance tests for N
+				// iterations to identify flaking
+				// tests. This should only be set to
+				// true in development (i.e., not CI).
+				//
+				// This should be run every time we
+				// consume any new version of haproxy
+				// be that a major, minor or a micro
+				// update to continuously validate the
+				// set of test case IDs that fail.
+				if sampleForFailures := false; sampleForFailures {
+					runConformanceTestsAndLogAggregateFailures(oc, t, 50)
+					continue
+				}
+
+				testSuites := runConformanceTests(oc, t)
+				failures := failingTests(testSuites)
+				failureCount := len(failures)
+
+				g.By("Analyzing results")
+				for _, f := range failures {
+					if _, exists := t.knownFailures[f.ID()]; exists {
+						failureCount -= 1
+						e2e.Logf("[%s] TestCase ID: %q is a known failure; ignoring", t.routeType, f.ID())
+					} else {
+						e2e.Logf("[%s] TestCase ID: %q (%q) ****FAILED****", t.routeType, f.ID(), f.TestCase.ClassName)
+					}
+				}
+				o.Expect(failureCount).Should(o.BeZero(), "expected zero failures")
+			}
+		})
+	})
+})
+
+func failingTests(testSuites []*h2spec.JUnitTestSuite) []h2specFailingTest {
+	var failures []h2specFailingTest
+
+	for _, ts := range testSuites {
+		for i := 0; i < ts.Tests; i++ {
+			if ts.TestCases[i].Error != nil {
+				failures = append(failures, h2specFailingTest{
+					TestNumber: i + 1,
+					TestCase:   ts.TestCases[i],
+				})
+			}
+		}
+	}
+
+	return failures
+}
+
+func runConformanceTests(oc *exutil.CLI, t h2specRouteTypeTest) []*h2spec.JUnitTestSuite {
+	podName := "h2spec"
+	e2e.ExpectNoError(oc.KubeFramework().WaitForPodRunning(podName))
+
+	outputFile, err := ioutil.TempFile("/tmp", "runConformanceTests")
```

Comment:
Not sure if there are any gotchas writing a temp file in this context, might want to ask around. I would at least suggest `ioutil.TempFile("", "runConformanceTests")` for portability (which will use the system's default tmp dir)